src/main/java/org/jboss/tm/TransactionLocal.java

```diff
@@ -26,296 +26,257 @@
+     *
+     * <p>WARN: The current implemention just "locks the transactions"
+     *
+     * @throws IllegalStateException if the transaction is not active
+     * @throws InterruptedException if the thread is interrupted
+     */
+    public void lock() throws InterruptedException {
+        lock(getTransaction());
+    }
+
+    /**
+     * Lock the TransactionLocal using the provided transaction
+     *
+     * <p>
+     *
+     * <p>WARN: The current implemention just "locks the transactions"
+     *
+     * @param transaction the transaction
+     * @throws IllegalStateException if the transaction is not active
+     * @throws InterruptedException if the thread is interrupted
+     */
+    public void lock(Transaction transaction) throws InterruptedException {
+        // ignore when there is no transaction
+        if (transaction == null) return;
+
+        delegate.lock(this, transaction);
+    }
+
+    /** Unlock the TransactionLocal using the current transaction */
+    public void unlock() {
+        unlock(getTransaction());
+    }
+
+    /**
+     * Unlock the ThreadLocal using the provided transaction
+     *
+     * @param transaction the transaction
+     */
+    public void unlock(Transaction transaction) {
+        // ignore when there is no transaction
+        if (transaction == null) return;
+
+        delegate.unlock(this, transaction);
+    }
+
+    /**
+     * Returns the initial value for this thransaction local. This method will be called once per
+     * accessing transaction for each TransactionLocal, the first time each transaction accesses the
+     * variable with get or set. If the programmer desires TransactionLocal variables to be
+     * initialized to some value other than null, TransactionLocal must be subclassed, and this
+     * method overridden. Typically, an anonymous inner class will be used. Typical implementations
+     * of initialValue will call an appropriate constructor and return the newly constructed object.
+     *
+     * @return the initial value for this TransactionLocal
+     */
+    protected Object initialValue() {
+        return null;
+    }
+
+    /**
+     * get the transaction local value.
+     *
+     * @param tx the transaction
+     * @return the obejct
+     */
+    protected Object getValue(Transaction tx) {
+        return delegate.getValue(this, tx);
+    }
+
+    /**
+     * put the value in the TransactionImpl map
+     *
+     * @param tx the transaction
+     * @param value the value
+     */
+    protected void storeValue(Transaction tx, Object value) {
+        delegate.storeValue(this, tx, value);
+    }
+
+    /**
+     * does Transaction contain object?
+     *
+     * @param tx the transaction
+     * @return true if it has an object
+     */
+    protected boolean containsValue(Transaction tx) {
+        return delegate.containsValue(this, tx);
+    }
+
+    /**
+     * Returns the value of this TransactionLocal variable associated with the thread context
```

Comment:
The side by diffs between the old and new are diverging and I am struggling to compare the two.
So I am likely to miss quite a lot from this point on. If you can take what I've suggested so far but apply it throughout the PR then that may help. And if other team members can also take a look then that might catch what I'll inevitably miss, I think only @marcosgopen is around at the moment.

Reply:
> The side by diffs between the old and new are diverging and I am struggling to compare the two. So I am likely to miss quite a lot from this point on. If you can take what I've suggested so far but apply it throughout the PR then that may help. And if other team members can also take a look then that might catch what I'll inevitably miss, I think only @marcosgopen is around at the moment.

You are right, I suggest to clone my branch and comapre in Intellij